Storage/rest.h

```diff
@@ -0,0 +1,201 @@
+    auto& storage = this->storage;  // For lambdas.
+    return server.Register(
+        "/api/" + storage(::current::storage::FieldNameByIndex<INDEX>()),
+        URLPathArgs::CountMask::None | URLPathArgs::CountMask::One,
+        [&storage](Request r) {
+          if (r.method == "GET") {
+            if (r.url_path_args.size() != 1) {
+              r("TBD ERROR", HTTPResponseCode.BadRequest);
+            } else {
+              const auto& key = r.url_path_args[0];
+              storage.Transaction([key, &storage](ImmutableFields<STORAGE>) -> Response {
+                return RESTfulGETWrapper<INDEX, typename ENTRY_TYPE::type>::RunGET(storage, key);
+              }, std::move(r)).Wait();
+            }
+          } else if (r.method == "POST") {
+            if (!r.url_path_args.empty()) {
+              r("TBD ERROR", HTTPResponseCode.BadRequest);
+            } else {
+              try {
+                const auto body = ParseJSON<typename ENTRY_TYPE::type>(r.body);
+                storage.Transaction([body, &storage](MutableFields<STORAGE>) {
+                  RESTfulPOSTWrapper<INDEX, typename ENTRY_TYPE::type>::RunPOST(storage, body);
+                }).Wait();
+                r("", HTTPResponseCode.NoContent);
+              } catch (const TypeSystemParseJSONException&) {
+                r("TBD ERROR", HTTPResponseCode.BadRequest);
+              }
+            }
+          } else if (r.method == "DELETE") {
+            if (r.url_path_args.size() != 1) {
+              r("TBD ERROR", HTTPResponseCode.BadRequest);
+            } else {
+              const auto& key = r.url_path_args[0];
+              storage.Transaction([key, &storage](MutableFields<STORAGE>) {
+                RESTfulDELETEWrapper<INDEX, typename ENTRY_TYPE::type>::RunDELETE(storage, key);
+              }).Wait();
+              r("", HTTPResponseCode.NoContent);
+            }
+          } else {
+            r("", HTTPResponseCode.MethodNotAllowed);
+          }
+        });
+  }
+};
+
+template <int INDEX, typename SERVER, typename STORAGE>
+HTTPRoutesScopeEntry RegisterRESTfulStorageEndpoint(SERVER& server, STORAGE& storage) {
+  return storage(::current::storage::FieldNameAndTypeByIndexAndReturn<INDEX, HTTPRoutesScopeEntry>(),
+                 RESTfulStorageEndpointRegisterer<INDEX, SERVER, STORAGE>(server, storage));
+}
+
+template <typename>
+struct RegisterRESTfulStorageEndpoints;
+
+template <int N, int... NS>
+struct RegisterRESTfulStorageEndpoints<current::variadic_indexes::indexes<N, NS...>> {
+  template <typename T_SERVER, typename T_STORAGE>
+  static void Run(HTTPRoutesScope& scope, T_SERVER& http_server, T_STORAGE& storage) {
+    scope += RegisterRESTfulStorageEndpoint<N>(http_server, storage);
+    RegisterRESTfulStorageEndpoints<current::variadic_indexes::indexes<NS...>>::Run(
+        scope, http_server, storage);
+  }
+};
+
+template <>
+struct RegisterRESTfulStorageEndpoints<current::variadic_indexes::indexes<>> {
```

Comment:
Looks like a simple `template<int I, int N>` calling `<I+1, N>`, and having a corner case `<N, N>` would do here, no need in `variadic_indexes`.